hazelcast-jet-kafka/src/main/java/com/hazelcast/jet/impl/connector/kafka/StreamKafkaP.java

```diff
@@ -91,36 +96,45 @@
     protected void init(@Nonnull Context context) throws Exception {
         snapshottingEnabled = context.snapshottingEnabled();
         consumer = new KafkaConsumer<>(properties);
-        reassignPartitions();
+        reassignPartitions(true);
     }
 
-    private void reassignPartitions() {
-        boolean changed = false;
+    private void reassignPartitions(boolean isInitial) {
+        List<TopicPartition> addedPartitions = new ArrayList<>();
         // check for added partitions (kafka doesn't support partition removal). Initially, all partitions are added.
         for (int topicIdx = 0; topicIdx < topicIds.size(); topicIdx++) {
             String topicName = topicIds.get(topicIdx);
             List<PartitionInfo> partitionInfos = consumer.partitionsFor(topicName);
-            getLogger().finest("Num of ptions for topic " + topicName + ": " + partitionInfos.size());
-            while (nextAssignablePtions[topicIdx] < partitionInfos.size()) {
-                int partition = nextAssignablePtions[topicIdx];
-                nextAssignablePtions[topicIdx] += processorCount;
-                assert partitionInfos.get(partition).partition() == partition;
-                assignment.add(new TopicPartition(topicName, partition));
-                changed = true;
+            LoggingUtil.logFinest(getLogger(), "Num of ptions for topic '%s': %d", topicName, partitionInfos.size());
+            if (nextAssignablePtions[topicIdx] < partitionInfos.size()) {
```

Comment:
Here you can reduce the nesting level by inverting the `if` and using `continue`.